tools/install_ssl_configuration.sh

```diff
@@ -0,0 +1,320 @@
+echo "OpenSearch config file: $OPENSEARCH_CONF_FILE"
+echo "OpenSearch bin dir: $OPENSEARCH_BIN_DIR"
+echo "OpenSearch plugins dir: $OPENSEARCH_PLUGINS_DIR"
+echo "OpenSearch lib dir: $OPENSEARCH_LIB_PATH"
+echo "Detected OpenSearch Version: $OPENSEARCH_VERSION"
+echo "Detected OpenSearch Security Version: $SECURITY_VERSION"
+
+if $SUDO_CMD grep --quiet -i plugins.security "$OPENSEARCH_CONF_FILE"; then
+  echo "$OPENSEARCH_CONF_FILE seems to be already configured for Security. Quit."
+  exit $skip_updates
+fi
+
+set +e
+
+set -e
+
+#echo "$ADMIN_CERT" | $SUDO_CMD tee "$OPENSEARCH_CONF_DIR/kirk.pem" > /dev/null
+#echo "$NODE_CERT" | $SUDO_CMD tee "$OPENSEARCH_CONF_DIR/esnode.pem" > /dev/null
+#echo "$ROOT_CA" | $SUDO_CMD tee "$OPENSEARCH_CONF_DIR/root-ca.pem" > /dev/null
+#echo "$NODE_KEY" | $SUDO_CMD tee "$OPENSEARCH_CONF_DIR/esnode-key.pem" > /dev/null
+#echo "$ADMIN_CERT_KEY" | $SUDO_CMD tee "$OPENSEARCH_CONF_DIR/kirk-key.pem" > /dev/null
+cd "$OPENSEARCH_CONF_DIR"
+if ([ "$use_defaults" == 1 ]); then
+    openssl genrsa -out root-ca-key.pem 2048
+    openssl req -new -x509 -sha256 -key root-ca-key.pem -subj "/C=US/ST=WA/L=SEATTLE/O=/OU=/CN=ROOT" -out root-ca.pem
+    # Admin cert
+    openssl genrsa -out kirk-key-temp.pem 2048
+    openssl pkcs8 -inform PEM -outform PEM -in kirk-key-temp.pem -topk8 -nocrypt -v1 PBE-SHA1-3DES -out kirk-key.pem
+    openssl req -new -key kirk-key.pem -subj "/C=US/ST=WA/L=SEATTLE/O=/OU=/CN=KIRK" -out kirk.csr
+    openssl x509 -req -in kirk.csr -CA root-ca.pem -CAkey root-ca-key.pem -CAcreateserial -sha256 -out kirk.pem
+    # Node cert
+    openssl genrsa -out esnode-key-temp.pem 2048
+    openssl pkcs8 -inform PEM -outform PEM -in esnode-key-temp.pem -topk8 -nocrypt -v1 PBE-SHA1-3DES -out esnode-key.pem
+    openssl req -new -key esnode-key.pem -subj "/C=US/ST=WA/L=SEATTLE/O=/OU=/CN=ESNODE" -out esnode.csr
+    openssl x509 -req -in esnode.csr -CA root-ca.pem -CAkey root-ca-key.pem -CAcreateserial -sha256 -out esnode.pem
+else
+    read -r -p "How many days should keys be active for? " response
+    openssl genrsa -out root-ca-key.pem 2048
+    openssl req -new -x509 -sha256 -key root-ca-key.pem -out root-ca.pem -days $response
+    # Admin cert
+    openssl genrsa -out kirk-key-temp.pem 2048
+    openssl pkcs8 -inform PEM -outform PEM -in kirk-key-temp.pem -topk8 -nocrypt -v1 PBE-SHA1-3DES -out kirk-key.pem    
+    openssl req -new -key kirk-key.pem -out kirk.csr
+    openssl x509 -req -in kirk.csr -CA root-ca.pem -CAkey root-ca-key.pem -CAcreateserial -sha256 -out kirk.pem -days $response
+    # Node cert
+    openssl genrsa -out esnode-key-temp.pem 2048
+    openssl pkcs8 -inform PEM -outform PEM -in esnode-key-temp.pem -topk8 -nocrypt -v1 PBE-SHA1-3DES -out esnode-key.pem
+    openssl req -new -key esnode-key.pem -out esnode.csr
+    openssl x509 -req -in esnode.csr -CA root-ca.pem -CAkey root-ca-key.pem -CAcreateserial -sha256 -out esnode.pem -days $response
+fi
```

Comment:
if and else section do the same thing with different inputs. Code duplication can be avoided by abstracting this in a function and invoking that